test/instrumentation/modules/@aws-sdk/s3-v3.test.js

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict'
+
+if (process.env.GITHUB_ACTIONS === 'true' && process.platform === 'win32') {
+  console.log('# SKIP: GH Actions do not support docker services on Windows')
+  process.exit(0)
+}
+
+// Test S3 instrumentation of the '@aws-sdk/client-s3' module.
+//
+// Note that this uses localstack for testing, which mimicks the S3 API but
+// isn't identical. Some known limitations:
+// - It basically does nothing with regions, so testing bucket region discovery
+//   isn't possible.
+// - AFAIK localstack does not support Access Points, so access point ARNs
+//   cannot be tested.
+
+const { execFile } = require('child_process')
+const util = require('util')
+
+const tape = require('tape')
+
+const { MockAPMServer } = require('../../../_mock_apm_server')
+const { validateSpan } = require('../../../_validate_schema')
+
+const LOCALSTACK_HOST = process.env.LOCALSTACK_HOST || 'localhost'
+const endpoint = 'http://' + LOCALSTACK_HOST + ':4566'
+
+// Execute 'node fixtures/use-s3js' and assert APM server gets the expected
+// spans.
+tape.test('simple S3 V3 usage scenario', function (t) {
+  const server = new MockAPMServer()
+  server.start(function (serverUrl) {
+    const additionalEnv = {
+      ELASTIC_APM_SERVER_URL: serverUrl,
+      AWS_ACCESS_KEY_ID: 'fake',
+      AWS_SECRET_ACCESS_KEY: 'fake',
+      TEST_BUCKET_NAME: 'elasticapmtest-bucket-1',
```

Comment:
```suggestion
      TEST_BUCKET_NAME: 'elasticapmtest-bucket-3',
```

Let's avoid re-using the same bucket name from s3.test.js in case we ever get into running test files in parallel -- which I hope we do at some point.